scripts/generate_asm.py

```diff
@@ -0,0 +1,73 @@
+import sys
+import re
+
+def replace_asm(file):
+    """
+    Creates all the possible files with all the combinations of the replacements
+    """
+    with open(file, 'r') as f:
+        lines = f.readlines()
+    replacements = get_replacements(lines)
+    files_to_replace = [lines.copy()]
+    
+    for str_to_replace in replacements:
+        new_files_to_replace = []
+        for _file in files_to_replace:
+            new_files = replace_once(_file,str_to_replace)
+            new_files_to_replace.extend(new_files)
+        if len(new_files_to_replace) != 0:
+            files_to_replace = new_files_to_replace
+
+    for j,_file in enumerate(files_to_replace):
+        with open(file[:-5] + "_replaced_" + str(j) + ".zasm", 'w') as f:
```

Comment:
Let's make it so the script generates the files into the tests programs folder, or accept an argument with the output folder, whichever you prefer.


Ideally, we should be able to just do make 'test_template.asm' and generate the tests files.